src/model_scoring_functions.R

```diff
@@ -0,0 +1,219 @@
+#' Functions to calculate energy scores for model output files according to given reference date
+#'
+#' @param hub_path character string, path to the root of the hub from the current working directory,
+#' defaults to assume that variant-nowcast-hub/src is the working directory
+#' @param model_output_file character string, directory under variant-nowcast-hub/[team name]/[model output parquet file]
+#' @param ref_date character string, date corresponding to model submission deadline, also called nowcast date
+#'
+#' @examples
+#' df_score <- get_energy_scores(hub_path = "../", model_output_file = "UMass-HMLR/2024-10-16-UMass-HMLR.parquet", 
+#' ref_date = "2024-10-16")
+get_energy_scores <- function(
+    hub_path = "../",
+    model_output_file = NULL,
+    ref_date = NULL
+){
+  require("tidyr")
+  require("dplyr")
+  require("readr")
+  require("arrow")
+  require("scoringRules")
```

Comment:
A minor note, but may be more work than its worth:

loading packages in a function is not good practice and may cause headaches down the line (i.e. dplyr masks some functions from stats and base).

I'd be happy to help rectify this.